clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -177,14 +177,14 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
                             // move partition from the member with an older generation to the member with the newer generation
                             ownedPartitions.add(tp);
                             consumerToOwnedPartitions.get(otherConsumer).remove(tp);
-                            allPreviousPartitionsToOwner.put(tp, consumer);
                             log.warn("Consumer {} in generation {} and consumer {} in generation {} claiming the same " +
                                             "TopicPartition {} in different generations. The topic partition will be " +
                                             "assigned to the member with the higher generation {}.",
                                     consumer, memberGeneration,
                                     otherConsumer, otherMemberGeneration,
                                     tp,
                                     memberGeneration);
+                            allPreviousPartitionsToOwner.put(tp, consumer);
```

Comment:
Does moving this line after the log.warn do anything?

Reply:
Agreed we should move it back to the original position

Reply:
done